### PR TITLE
USHIFT-1320 USHIFT-1431: E2E RF Test: "block downgrade"

### PR DESCRIPTION
**Pull request by pmtk**

### CODE REVIEW

File: test/resources/ostree.resource [hunk 1]
```diff
@@ -30,15 +23,15 @@ Current Deployment Should Be
...
     [Documentation]    Deploys given ref and configures test agent for failing greenboot.
     ...    It expects the system to roll back.
-    [Arguments]    ${ref}
+    [Arguments]    ${ref}    ${write_agent_cfg}=${TRUE}
 
     ${initial_deploy_id}=    Get Booted Deployment Id
     ${deploy_id}=    Rebase System    ${ref}
-    TestAgent.Write
-    Write Greenboot Microshift Wait Timeout    60
+    IF    ${write_agent_cfg}    TestAgent.Write
```
Comment (dhellmann) [inline test/resources/ostree.resource:30]: Is the config always the same?

File: test/scenarios/rhel-9.2-microshift-source-block-downgrade.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# Sourced from scenario.sh and uses functions defined there.
+
+scenario_create_vms() {
+    prepare_kickstart host1 kickstart.ks.template rhel-9.2-microshift-source-fake-next-minor
```
Comment (dhellmann) [inline test/scenarios/rhel-9.2-microshift-source-block-downgrade.sh:6]: We could combine the next-minor and previous-minor images for a similar test that ensures we block skipping 2 minor versions.

**Review by dhellmann — COMMENTED** [review]
/lgtm

File: test/scenarios/rhel-9.2-microshift-source-block-downgrade.sh (hunk 2 above)
Comment (pmtk) [reply test/scenarios/rhel-9.2-microshift-source-block-downgrade.sh:6]: This is very good point and I have a feeling that we lost implementation of this when upgrading from 4.13... https://github.com/openshift/microshift/blob/main/pkg/admin/prerun/version.go#L36

File: test/resources/ostree.resource (hunk 1 above)
Comment (pmtk) [reply test/resources/ostree.resource:30]: No, this line is only persist what test should set up, see [fdo.robot#L33](https://github.com/openshift/microshift/blob/main/test/suites-ostree/fdo.robot#L33) and [failed-upgrade.robot#L30](https://github.com/openshift/microshift/blob/main/test/suites-ostree/failed-upgrade.robot#L30)

Comment (dhellmann) [reply test/resources/ostree.resource:30]: Ah, right, I forgot the config was managed separately from the write step.

File: test/scenarios/rhel-9.2-microshift-source-block-downgrade.sh (hunk 2 above)
Comment (dhellmann) [reply test/scenarios/rhel-9.2-microshift-source-block-downgrade.sh:6]: We could attach a test to https://issues.redhat.com/browse/USHIFT-1088

**Pull request merged**